### ISSUE TITLE
BoringSSL Compilation Error: implict fallthrough

### ISSUE THREAD

**Issue by rhclayto** [grpc/grpc#12536]
When trying to build grpc with GCC 7, the installation always fails for me during the BoringSSL compilation.

GCC 7 now throws an error when it finds an 'implicit fallthrough' in the C code (https://developers.redhat.com/blog/2017/03/10/wimplicit-fallthrough-in-gcc-7/). The BoringSSL that comes with grpc has an implicit fallthrough in `case BIO_CTRL_RESET` in `third_party/boringssl/crypto/bio/fd.c`:

```c
case BIO_CTRL_RESET:
  num = 0;
case BIO_C_FILE_SEEK:
 ...
```

The compilation error caused by this can be remedied on grpc's side by adding the following to the boringssl target in `binding.gyp`:

```
'cflags': [
  '-Wimplicit-fallthrough=0'
],
```

Hope this helps.

**Comment by vbatts** [comment]
I've opened an upstream changeset for this with boringssl: https://boringssl-review.googlesource.com/c/boringssl/+/20564

But the patch for the submodule'd version of boringssl that grpc master (eb44079fcd452e2f94066ffe0944c1390987b45e) has right now is: https://pastebin.com/br0DugFt

**Comment by PowerInside** [comment]
Reopen this issue? Because the boringssl submodule needs to be updated and the npm module version bumped for the issue to be completely resolved, right? grpc 1.6.6 for Node.js still has this issue.

Related PR: https://github.com/grpc/grpc/pull/12922
